hw02-DIYarrayList/src/main/java/hw02/MyArrayList.java

```diff
@@ -0,0 +1,230 @@
+package hw02;
+
+import java.lang.IndexOutOfBoundsException;
+import java.lang.UnsupportedOperationException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+
+public class MyArrayList<T> implements List<T> {
+    private static final int INITIAL_CAPACITY = 30;
+
+    private Object[] _array;
+    private int numberOfElements = 0;
+
+    public MyArrayList() {
+        _array = new Object[INITIAL_CAPACITY];
+    }
+
+    public T set(int index, T element) {
+        if(index < 0 || index > size()) {
+            throw new IndexOutOfBoundsException();
+        }
+
+        T prevValue = (T)_array[index];
+        _array[index] = element;
+
+        return prevValue;
+    }
+
+    public T get(int index) {
+        if(index < 0 || index >= size()) {
+            throw new IndexOutOfBoundsException();
+        }
+
+        return (T)_array[index];
+    }
+
+    public boolean add(T e) {
+        if(isFull()) {
+            grow();
+        }
+
+        _array[numberOfElements++] = e;
+
+        return true;
+    }
+
+    public Object[] toArray() {
+        return Arrays.copyOf(_array, size());
+    }
+
+    public Iterator<T> iterator() {
+        return new Itr();
+    }
+
+    public boolean isEmpty() {
+        return _array.length == 0;
+    }
+
+    public int size() {
+        return _array.length;
+    }
+
+    public void printList() {
+        for(int i = 0; i < size(); i++) {
+            if(i != 0) {
+                System.out.print(",");
+            }
+
+            System.out.print(get(i));
+        }
+
+        System.out.println();
+    }
+
+    public void growToSize(int requiredSize) {
```

Comment:
А почему public? Не надо public. Это чисто внутренности реализации.
Кто-нибудь снаружи вызовет этот метод и сломает что-нибудь.

```suggestion
    private void growToSize(int requiredSize) {
```